smap/sources/aqiCalculator.py

```diff
@@ -0,0 +1,96 @@
+import requests
+import simplejson as json
+
+# used for logging
+import logging
+import sys, traceback
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
+BASE_URL = "http://localhost/raqmn_api/"
+
+def init_aqi_drivers():
+	"""
+	Used to initialize AQI Driver
+	Returns all the dylos sensors available with their serial number and uuid
+	"""
+	try:
+		data = json.loads(requests.get(BASE_URL).content)
+		dylosSensors = {}
+		for datum in data:
+			try:
+				pathParts = datum['Path'].split("/") 
+				if len(pathParts)==4 and pathParts[1].strip()=="dylos":
+					dylosSensors[pathParts[2].strip()+"/"+pathParts[3].strip()] = datum['uuid']
+					get_aqi_data(1, datum['uuid'],pathParts[3].strip())
+			except:
+				logging.debug("Unexpected error : "+ str(sys.exc_info()[0]))
+				logging.debug("Traceback : "+ str(traceback.format_exc()))
+		logging.debug(dylosSensors)
+		return dylosSensors
+	except:
+		logging.debug("Unexpected error : "+ str(sys.exc_info()[0]))
+		logging.debug("Traceback : "+ str(traceback.format_exc()))
+        return {}
+
+def get_aqi_data(hours,uuid,type):
+	"""
+	Returns the AQI for number of hours given for the sensor mentioned by uuid
+	"""
+	try:
+		data = json.loads(requests.get(BASE_URL+"data/"+str(hours)+"h/"+uuid+"/").content)
+		dataPoints = remove_outliers([x[1] for x in data])
+		average = float(sum(dataPoints))/len(dataPoints) if len(dataPoints) > 0 else 0
+		aqi = convert_particle_count_to_aqi(average, type)
+		logging.debug(aqi)
+		return aqi
+	except:
+		logging.debug("Unexpected error : "+ str(sys.exc_info()[0]))
+		logging.debug("Traceback : "+ str(traceback.format_exc()))
+        return 0
+
+def remove_outliers(data):
+	"""
+	Removes outliers from timeseries data given
+	TODO: This function uses a very simple trimming algorithm for removing outliers. Need a more robust algorithm for timeseries data to remove outliers
+	"""
+	data.sort()
+	numPointsToRemove = (5*len(data))/100
+	return data[numPointsToRemove:len(data)-numPointsToRemove]
+
+def convert_particle_count_to_aqi(average, type):
```

Comment:
I didn't check this function since it is basically a bunch of math formulae and I trust that you reproduced them correctly from the references. Maybe add a simple unit test that does the calculation for known values and checks that the known result is obtained?

I want to make sure that we are not off by a factor or 10 or something.